tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -1566,20 +1566,47 @@ Status BinaryTensorOpWeight(OpConverterParams* params,
   return tensorflow::Status::OK();
 }
 
-enum class ConvolutionType { DEFAULT, DEPTHWISE_CONV };
+enum class ConvolutionType { DEFAULT, DEPTHWISE_CONV, TRANSPOSE };
 
-tensorflow::Status ConvertConv2DHelper(OpConverterParams* params, int group) {
+tensorflow::Status ConvertConv2DHelper(OpConverterParams* params, int group,
+                                       bool is_deconv) {
   const auto& inputs = params->inputs;
   const auto& node_def = params->node_def;
-  if (inputs.size() != 2) {
-    return tensorflow::errors::InvalidArgument("Two inputs are expected for ",
-                                               node_def.op(), ", at ",
-                                               node_def.name());
-  }
-  if (inputs.at(0).is_weights()) {
-    return tensorflow::errors::Unimplemented(
-        node_def.op(), " is only implemented for tensors, not weights, at ",
-        node_def.name());
+  TRT_TensorOrWeights input, deconv_output_size;
+  if (is_deconv) {
```

Comment:
optional: we can merge both branches to make it concise by setting conditionally setting a vector of is_tensor and using that for the check on the number of inputs and their types between weights and tensors.

Reply:
That is a good idea. Perhaps we have can some helper functions to validate the inputs like you said - this would help reduce code duplication. I can do this in a separate PR.

Reply:
Sure. Please add this note as a TODO.